pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -480,7 +481,9 @@ protected void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
         long totalMessagesSent = 0;
         long totalBytesSent = 0;
 
-        while (entriesToDispatch > 0 && totalAvailablePermits > 0 && isAtleastOneConsumerAvailable()) {
+        int firstAvailableConsumerPermits = getFirstAvailableConsumerPermits();
+        int currentTotalAvailablePermits = Math.max(totalAvailablePermits, firstAvailableConsumerPermits);
+        while (entriesToDispatch > 0 && currentTotalAvailablePermits > 0 && firstAvailableConsumerPermits > 0) {
```

Comment:
isn't there a difference in behavior here now that  `firstAvailableConsumerPermits > 0` is evaluated once before the while loop and previously `isAtleastOneConsumerAvailable()` would get evaluated each time the while loop evaluates the condition?

Reply:
perhaps something like `(firstAvailableConsumerPermits > 0 || isAtleastOneConsumerAvailable())` ?

Reply:
thanks for catching it. yes, I missed to push the commit. fixed now.